core/src/main/scala/pink/cozydev/protosearch/IndexSearcher.scala

```diff
@@ -124,6 +126,19 @@ object IndexSearcher {
             case _ => Left("Unsupport TermRange error?")
           }
       }
+
+    private def regexSearch(q: Query.TermRegex): Either[String, Set[Int]] =
+      try {
+        val regex = q.str.r
+        val terms = index.termDict.termsForRegex(regex)
+        Right(
+          terms
+            .flatMap(m => index.docsWithTerm(m))
+            .toSet
+        )
+      } catch {
+        case _: PatternSyntaxException => Left(s"Invalid regex query $q")
+      }
```

Comment:
This works but it puts a lot more stuff into the `try` block than just the regex creation.
Can we isolate the `try` block to just the regex creation?
Maybe something like:

```scala
val regex: Either[String, Regex] =
  try {
    Right(q.str.r)
  } catch {
    case _: PatternSyntaxException => Left(s"Invalid regex query $q")
  }
```


Admittedly, part of me is wondering if there should be a `toRegex` method on `Query.TermRegex` itself...
But we can save that for another issue.

Reply:
Makes sense, I was a doubtful about the large try-catch block.

I pushed code with a slight modification to the try-catch where instead of setting the type for regex as Either[String, Regex] it just return the error message for the entire function as it seemed redundant to match for the error and just return it unmodified. Let me know if I should avoid doing that for code readability, it's a quick change on my end.

Reply:
Like this
```
      val regex =
        try
          q.str.r
        catch {
          case _: PatternSyntaxException => return Left(s"Invalid regex query $q")
        }      
```